File: backend/app/02_medium_complexity/src/store/themeSlice.py
from typing import Any, Dict



class ThemeState:
    theme: Theme
const initialState: \'ThemeState\' = {
  theme: 'light',
}
const themeSlice = createSlice({
  name: 'theme',
  initialState,
  reducers: Dict[str, Any],
    toggleTheme: (state: ThemeState) => {
      state.theme = state.theme === 'light' ? 'dark' : 'light'
    },
  },
})
const { setTheme, toggleTheme } = themeSlice.actions
const themeReducer = themeSlice.reducer 

File: src/store/themeSlice.ts
import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { Theme } from '../types';

interface ThemeState {
  theme: Theme;
}

const initialState: ThemeState = {
  theme: 'light',
};

const themeSlice = createSlice({
  name: 'theme',
  initialState,
  reducers: {
    setTheme: (state: ThemeState, action: PayloadAction<Theme>) => {
      state.theme = action.payload;
    },
    toggleTheme: (state: ThemeState) => {
      state.theme = state.theme === 'light' ? 'dark' : 'light';
    },
  },
});

export const { setTheme, toggleTheme } = themeSlice.actions;
export const themeReducer = themeSlice.reducer; 